src/main/cpp/blaze.cc

```diff
@@ -1423,14 +1423,28 @@ static map<string, EnvVarValue> PrepareEnvironmentForJvm() {
   if (!blaze::GetEnv("http_proxy").empty() ||
       !blaze::GetEnv("HTTP_PROXY").empty()) {
     BAZEL_LOG(WARNING)
-        << "detected http_proxy set in env, setting no_proxy for localhost.";
+        << "detected HTTP_PROXY set in env, setting NO_PROXY for localhost.";
 
     // Disable HTTP proxies for localhost and any localhost-like address,
     // in case we (or gRPC, etc.) ever use one of these addresses.
-    std::string localhost_addresses = "localhost,127.0.0.1,0:0:0:0:0:0:0:1,::1";
+    string localhost_addresses = blaze::GetEnv("NO_PROXY");
```

Comment:
This seems wrong now... 

If we are going to extend the existing no_proxy and NO_PROXY values in the environment, then we should probably set result["no_proxy"] and result["NO_PROXY"] independently, instead of setting both of them to be the same value.

(In case no_proxy and NO_PROXY are both already set, to different non-empty values).

----

An alternative to setting these independently might be to check if no_proxy and NO_PROXY are both set to different non-empty values, and raise an error in that case. Similarly, we could raise an error if http_proxy and HTTP_PROXY are both set to different non-empty values.

Reply:
This is a nice summary of the state of weirdness
https://unix.stackexchange.com/questions/212894/whats-the-right-format-for-the-http-proxy-environment-variable-caps-or-no-ca

I would opt for doing the warning and setting them to the same value. That will result in the least surprise for the user if some process invoked by the build uses a different version of the variable than Bazel does. It also allows us to avoid the debate by allowing either upper or lower case. We don't pick sides and say it must be done the way tool X does.

Throwing an error is also acceptable. When a user has them mismatched and both are not empty, then we have a few likely things
1. have cargo-culted settings until things "worked", updating NO_PROXY and no_proxy independently.
2. or they have precisely set the minimal amount needed. E.g. They don't want to proxy against 2 addresses, but one subtool uses only 1 of them and uses no_proxy, while the other uses NO_PROXY and uses both. It would be legitimate to have different setting but harmless to force them to the bigger set.
3. or they really need the Bazel instance to use separate proxy settings from those of some invoked tool. This is a bizarre use case. I'm blanking on any scenario where this makes sense and is not a latent portability problem.

Reply:
Actually, since this is only done in the method to prepare no_proxy for the JVM, we should raise an error if no_proxy/NO_PROXY are mismatched. The JVM is only going to use one of the values, so there is no reason you could expect having different values to do anything useful.